src/main/java/seedu/address/logic/commands/ShowCommand.java

```diff
@@ -0,0 +1,105 @@
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ":\n1. Displays details of the person with given name, "
+            + "the specified keywords (case-insensitive)\n"
+            + "Parameters: KEYWORD\n"
+            + "Example: " + COMMAND_WORD + " alice\n"
+            + "2. Displays details of the person at the given index\n"
+            + "Parameters : INDEX\n"
+            + "Example: " + COMMAND_WORD + " 2";
+
+    public static final String MESSAGE_NO_NAME = "No User with Given Name could be found in the List";
+
+    public static final String MESSAGE_SUCCESS = "Showing details for %1$s.";
+
+    public static final String MESSAGE_MULTIPLE_NAME = "%1$d Persons with the given keyword in their name exists.\n"
+            + "You can try show with index or full name";
+
+    private final Predicate<Person> predicate;
+    private final Index index;
+
+    /**
+     * Constructor for Show Command when name is given
+     *
+     * @param predicate contains Persons with name matching keyword
+     */
+    public ShowCommand(Predicate<Person> predicate) {
+        this.predicate = predicate;
+        this.index = null;
+    }
+
+    /**
+     * Constructor for Show Command when index is given
+     *
+     * @param index of the Person to show
+     */
+    public ShowCommand(Index index) {
+        this.predicate = null;
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        FilteredList<Person> filteredList = new FilteredList<>(model.getFilteredPersonList());
+        if (predicate != null) {
+            filteredList.setPredicate(predicate);
+            if (filteredList.isEmpty()) {
+                filteredList = new FilteredList<>(model.getAddressBook().getPersonList());
+                filteredList.setPredicate(predicate);
+                if (filteredList.isEmpty()) {
+                    return new CommandResult(MESSAGE_NO_NAME);
+                } else {
+                    model.updateFilteredPersonList(model.PREDICATE_SHOW_ALL_PERSONS);
+                    this.execute(model);
+                }
+            }
+            if (filteredList.size() == 1) {
+                int index = model.getFilteredPersonList().indexOf(filteredList.get(0));
+                model.setSelectedIndex(index);
+                return new CommandResult(String.format(MESSAGE_SUCCESS,
+                        model.getFilteredPersonList().get(index).getName()));
+            } else {
+                model.updateFilteredPersonList(predicate);
+                return new CommandResult(String.format(MESSAGE_MULTIPLE_NAME, model.getFilteredPersonList().size()));
+            }
+        } else if (index != null) {
+            int index = this.index.getZeroBased();
+            if (index >= model.getFilteredPersonList().size()) {
+                return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
```

Comment:
I believe alternate error paths should be thrown as `new CommandException(errror message)`, to be handled in MainWindow.java. Maybe you could refer to the execute implementation in `EditCommand.java`

Reply:
Threw CommandException for invalid index and invalid name.